AutoRaise.mm

```diff
@@ -475,48 +475,140 @@ CGEventRef eventTapHandler(CGEventTapProxy proxy, CGEventType type, CGEventRef e
 }
 
 #define POLLING_MS 20
+#define VERSION "1.8"
 int main(int argc, const char * argv[]) {
+    printf("\nv%s by sbmpost(c) 2021, usage:\nAutoRaise -delay <1=%dms> [-warpX <0.5> -warpY <0.5> -scale <2.0>]", VERSION, POLLING_MS);
     @autoreleasepool {
-        if (argc >= 3) {
-            NSUserDefaults * standardDefaults = [NSUserDefaults standardUserDefaults];
-            delayCount = abs((int) [standardDefaults integerForKey: @"delay"]);
-            warpMouse = argc != 3 ;
-            if (argc >= 7) {
-                warpX = [standardDefaults floatForKey: @"warpX"];
-                warpY = [standardDefaults floatForKey: @"warpY"];
-                if (argc >= 8) {
-                    cursorScale = [standardDefaults floatForKey: @"scale"];
-                }
+        
+        #define defaultDelay @"delay"
+        #define defaultWarpX @"warpX"
+        #define defaultWarpY @"warpY"
+        #define defaultScale @"scale"
+        
+        // initial values for standardDefaults
+        NSDictionary* standardDefaults = @{defaultDelay:@-1,
+                                           defaultWarpX:@-1.0,
+                                           defaultWarpY:@-1.0,
+                                           defaultScale:@2.0 };
+        
+        // sync standardDefaults with command-line arguments
```

Comment:
This seems like a better way to set the defaults when compared to what I am doing (reading the fields first and then provide default values for them if we got nil/null objects). There are actually 4 sources of information to consider:

1 command line parameters
2 original file structure
3 new file structure
4 defaults

If we could apply this sync strategy in a smart way so that we respect the priority of 1 to 4, then the code will probably be cleaner indeed.

Reply:
there are some issues with detecting nil objects in standardDefaults - its all have value = 0, instead nil  :/
so I set definitely wrong initial value (-1) and it is now clear - is there any arguments from command line was received.

Priority of arguments same as you wrote:
1) valid command line arguments
2) original file structure
3) new files structure
4) dafults 

I'm sorry for my bad code :(

Reply:
Code is not too bad. In fact I also struggled with checking for the "existence of a certain key in the standardDefaults". I think you'd first need to make it an NSDictionary (standardDefaults has a method for that), and then you will have more flexibility.